src/functionalTest/groovy/com/autonomousapps/jvm/projects/JarTransformingProject.groovy

```diff
@@ -0,0 +1,113 @@
+package com.autonomousapps.jvm.projects
+
+import com.autonomousapps.AbstractProject
+import com.autonomousapps.kit.GradleProject
+import com.autonomousapps.kit.Plugin
+import com.autonomousapps.kit.Source
+import com.autonomousapps.kit.SourceType
+import com.autonomousapps.model.Advice
+import com.autonomousapps.model.ProjectAdvice
+
+import static com.autonomousapps.AdviceHelper.*
+import static com.autonomousapps.kit.Dependency.commonsCollections
+
+final class JarTransformingProject extends AbstractProject {
+
+  final GradleProject gradleProject
+
+  JarTransformingProject() {
+    this.gradleProject = build()
+  }
+
+  private def commonsCollections = commonsCollections("api")
+
+  private GradleProject build() {
+    def builder = newGradleProjectBuilder()
+    builder.withSubproject('proj') { s ->
+      s.sources = PROJ_SOURCES
+      s.withBuildScript { bs ->
+        bs.plugins = [Plugin.javaLibraryPlugin]
+        bs.sourceSets = ['integrationTest']
+        bs.dependencies = [
+          commonsCollections,
+        ]
+        bs.additions = '''
+          // We do some post processing of Jars on the classpath using a transform (split one Jar into multiple).
+          // A similar situation can also be created by publishing a component that has multiple Jars
+          // using Gradle Metadata to express that. 
+          def artifactType = Attribute.of("artifactType", String)
+          def split = Attribute.of("split", Boolean)
+          dependencies.attributesSchema { attribute(split) }
+          dependencies.artifactTypes.jar.attributes.attribute(split, false)
+          dependencies.registerTransform(SplitJarTransform) {
+            from.attribute(split, false).attribute(artifactType, "jar")
+            to.attribute(split, true).attribute(artifactType, "jar")
+          }
+          configurations.compileClasspath.attributes.attribute(split, false)
+          configurations.runtimeClasspath.attributes.attribute(split, false)
+          
+          afterEvaluate {
+            tasks.named("artifactsReportMain") {
```

Comment:
So, for the plugin to work in the context of this Gradle feature, users would have to manually configure this task?

Reply:
Well the situation can also be created differently. As I wrote in one comment: (one Variant of) one _Component_ can have several Jars. Other ways to produce this is to have published Metadata with more than one Jar in the _api_ (compile classpath) variant. Or by having a local project that, for whatever reason, produces more than one Jar for one variant. In these cases, the Jars will just be on the classpath and no adjustment would be needed to the _report_ task inputs.

(These are _mostly_ theoretical cases as in a default Java project, you won't do this. But you know how things go in large long-living projects....)

The test case here is motivated by what I have in a real project, where the Jars are produced by a _Transform_. The adjustment is needed because the _report_ task has an `artifactView` as input and not directly the `compileClasspath` Configuration. If you look closely at the setup, I reproduce the `artifactView` the task normally has as input, but add another attribute to it. You can attribute this to some of the _shortcomings_ of the rather complex _Transformations+ArtifactView_ APIs. Ideally, there would be some less invasive way to express this (somehow just adding the attribute - needed to trigger the transform - on top). But I don't know of another way to do this right now.

I will have a loot at adding a second case, where another local project produces the two Jars.

Reply:
b6bc3d484e525209c833599a830eda1450269e3a